### PR TITLE
MBS-10901 Fix LOST test incident definition

### PR DESCRIPTION
**Pull request by sergioboi**

### CODE REVIEW

File: docs/content/test_runner/Metrics.md [hunk 1]
```diff
@@ -96,6 +96,22 @@ These are cases when file was not found on device for some reason
 
 If test file was pulled, but there was a parsing error
 
+### `instr-parsing`
```
Comment (sergioboi) [inline docs/content/test_runner/Metrics.md:99]: Add concrete types of LOST

File: subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/internal/report/listener/ReportViewerTestReporter.kt [hunk 2]
```diff
@@ -60,84 +64,108 @@ internal class ReportViewerTestReporter(
...
-                    )
-                } catch (throwable: Throwable) {
-                    val (stdout: String, stderr: String) = logcatBuffers.getLogcat(key)
 
-                    val errorMessage = "Can't parse testRuntimeData: ${test.testName}; ${reportJson.readText()}"
+                        metricsSender.sendReportFileNotAvailable()
+                    }
+                )
+            }
+            is TestResult.InCompleted -> {
```
Comment (sergioboi) [inline subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/internal/report/listener/ReportViewerTestReporter.kt]: Before we tried to pull artifacts from the device even when test execution was `incomplete` and it led us to replace the original error.
Now we understand that test was `incomplete` and we don't try to pull artifacts from the device

File: subprojects/gradle/runner/client/src/main/kotlin/com/avito/runner/scheduler/listener/LogListener.kt [hunk 3]
```diff
@@ -30,14 +30,12 @@ internal class LogListener : TestListener {
             is TestCaseRun.Result.Ignored -> "IGNORED"
             is TestCaseRun.Result.Failed.InRun -> "FAILED"
             is TestCaseRun.Result.Failed.InfrastructureError -> {
-                device.logger.warn(result.errorMessage, result.cause)
+                device.logger.warn("${test.testName} LOST", result.error)
```
Comment (sergioboi) [inline subprojects/gradle/runner/client/src/main/kotlin/com/avito/runner/scheduler/listener/LogListener.kt:33]: Interesting that we already have a general place where we logging all test transitions

File: subprojects/gradle/runner/service/src/main/kotlin/com/avito/runner/service/model/TestCase.kt [hunk 4]
```diff
@@ -7,7 +7,7 @@ data class TestCase(
     val methodName: String,
     val deviceName: String
 ) : Serializable {
-    val testName: String = "$className.$methodName"
+    val testName: String = "$className.$methodName.$deviceName"
```
Comment (sergioboi) [inline subprojects/gradle/runner/service/src/main/kotlin/com/avito/runner/service/model/TestCase.kt:10]: I check all places where we use that property
Now we will differentiate tests by API

File: subprojects/gradle/runner/service/src/main/kotlin/com/avito/runner/service/model/TestCaseRun.kt [hunk 5]
```diff
@@ -15,10 +15,14 @@ data class TestCaseRun(
...
             data class InRun(val errorMessage: String) : Failed()
 
-            data class InfrastructureError(
-                val errorMessage: String,
-                val cause: Throwable? = null
-            ) : Failed()
+            sealed class InfrastructureError : Failed() {
+                abstract val error: Throwable
+
+                class Unexpected(override val error: Throwable) : InfrastructureError()
```
Comment (sergioboi) [inline subprojects/gradle/runner/service/src/main/kotlin/com/avito/runner/service/model/TestCaseRun.kt:21]: I like types

File: docs/content/test_runner/Metrics.md [hunk 6]
```diff
@@ -96,6 +96,22 @@ These are cases when file was not found on device for some reason
...
+
+### `instr-start`
+
+If fail when staring `am instrument`
+
+### `instr-timeout`
+
+If fail with timeout when executing `am instrument`
+
+### `unexpected`
```
Comment (dsvoronin) [inline docs/content/test_runner/Metrics.md]: add `instr-` for consistency?

File: subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/internal/report/listener/ReportViewerTestReporter.kt (hunk 2 above)
Comment (dsvoronin) [reply subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/internal/report/listener/ReportViewerTestReporter.kt]: Use `Incomplete` name: https://wikidiff.com/incomplete/incompleted

**Review by dsvoronin — APPROVED** [review]

File: subprojects/gradle/runner/service/src/main/kotlin/com/avito/runner/service/model/TestCase.kt (hunk 4 above)
Comment (eugene-krivobokov) [reply subprojects/gradle/runner/service/src/main/kotlin/com/avito/runner/service/model/TestCase.kt:10]: It seems that the value is used for debugging.
Maybe we should express it explicitly in API.
`testName` looks like a stricter contract.

File: subprojects/gradle/runner/client/src/main/kotlin/com/avito/runner/scheduler/listener/TestLifecycleListener.kt [hunk 7]
```diff
@@ -14,8 +15,13 @@ interface TestLifecycleListener {
...
     fun finished(
-        artifacts: Result<File>,
+        result: TestResult,
         test: TestCase,
         executionNumber: Int
     )
+
+    sealed class TestResult {
+        class Complete(val artifacts: Result<File>) : TestResult()
+        class InComplete(val infraError: InfrastructureError) : TestResult()
```
Comment (eugene-krivobokov) [inline subprojects/gradle/runner/client/src/main/kotlin/com/avito/runner/scheduler/listener/TestLifecycleListener.kt:25]:
```suggestion
        class Incomplete(val infraError: InfrastructureError) : TestResult()
```
If you mean "unfinished"

Comment (sergioboi) [reply subprojects/gradle/runner/client/src/main/kotlin/com/avito/runner/scheduler/listener/TestLifecycleListener.kt:25]: https://github.com/avito-tech/avito-android/pull/898

**Pull request merged**